[RNMobile] Activate preformat on android

marecar3 opened the pull request:
## Description
Activate `pre` format block on Android platform: https://github.com/wordpress-mobile/gutenberg-mobile/issues/1264

GB mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/1615

## How has this been tested?
Follow test instructions on `gb-mobile` PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/1615

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/native-mobile.md -->.

marecar3 committed:
Enable preformatted block on the Android and on production

marecar3 commented:
Changed target branch to `master` as we don't want to hurry with this feature (it still has some active issues)

hypest commented:
> Changed target branch to master

I think this PR also needs rebasing on top of master to only include its own commits. E.g. the "Spacer" commit is not really part of the PR.

![Screenshot 2019-11-28 at 15 25 24](https://user-images.githubusercontent.com/1032913/69809948-7b34bb80-11f3-11ea-827d-cfff7e611931.png)

marecar3 committed:
exclude pre and br tag from the html elements

marecar3 commented:
> > Changed target branch to master
> 
> I think this PR also needs rebasing on top of master to only include its own commits. E.g. the "Spacer" commit is not really part of the PR.
> 
> ![Screenshot 2019-11-28 at 15 25 24](https://user-images.githubusercontent.com/1032913/69809948-7b34bb80-11f3-11ea-827d-cfff7e611931.png)

Thanks, @hypest  I have fixed that one.

marecar3 committed:
in a case when tagName is pre return false in order to skip trim space checking

marecar3 committed:
Merge branch 'master' into rnmobile/activate_pre_format_android_new

marecar3 committed:
Fixed lint errors

hypest commented on packages/rich-text/src/component/index.native.js:
```diff
@@ -586,6 +586,15 @@ export class RichText extends Component {
+			tagName,
+		} = this.props;
+
+		// aztec won't trimm spaces in a case of <pre> block, so we are excluding it
```
Nit: typo in `trimm`, let's replace with `trim`.

hypest commented on packages/rich-text/src/component/index.native.js:
```diff
@@ -586,6 +586,15 @@ export class RichText extends Component {
+		} = this.props;
+
+		// aztec won't trimm spaces in a case of <pre> block, so we are excluding it
+		if ( tagName === 'pre' ) {
```
Nice and clear way of shortcircuiting `willTrimSpaces` in the Preformatted block case!

Can you also add a test in https://github.com/WordPress/gutenberg/blob/master/packages/rich-text/src/component/test/index.native.js to cover that case? Thanks!

marecar3 replied:
Good suggestion, thanks!

hypest reviewed (commented):
This works nicely! Let's add the unit test for the new `willTrimSpaces` codepath and fix the minor typo and I can review again!

marecar3 committed:
Fixed typo

marecar3 committed:
Add test case for preformatted block

marecar3 commented:
> This works nicely! Let's add the unit test for the new `willTrimSpaces` codepath and fix the minor typo and I can review again!

Hey @hypest, I have added unit test and also I have fixed minor typo. 
You can do another iteration of review :)

marecar3 committed:
Merge branch 'master' into rnmobile/activate_pre_format_android_new

marecar3 committed:
Fix lint errors

hypest commented:
👋 @marecar3 , seems that the PR needs updating from master and there are a couple of lint issues to fix too.

marecar3 committed:
Merge branch 'master' into rnmobile/activate_pre_format_android_new

# Conflicts:
#	packages/block-library/src/index.native.js

marecar3 commented:
> 👋 @marecar3 , seems that the PR needs updating from master and there are a couple of lint issues to fix too.

👋 @hypest , I have update this PR with the latest master and I am not sure about lint issues, as commit before this one was with ✅ https://github.com/WordPress/gutenberg/runs/336677088
but maybe I am missing something.

hypest commented:
Sorry, I think I didn't notice [the commit that fixed them before](https://github.com/WordPress/gutenberg/pull/18777/commits/3b4adb5f1831d0978685b1ee01c9fe761509425f)! Sorry for the confusion 😞

hypest reviewed (approved):
LGTM!

SergioEstevao reviewed (approved):
:shipit:
